docs/enhancements/ipxe-host-boot.md

```diff
@@ -0,0 +1,125 @@
+---
+title: ipxe-host-boot
+authors:
+  - "@carbonin"
+  - “@mhrivnak”
+creation-date: 2022-02-22
+last-updated: 2022-02-22
+---
+
+# iPXE Host Boot
+
+## Summary
+
+Not all user infrastructure is able to boot isos directly, and iPXE is a commonly 
+used alternative for booting machines. We should support this by making the 
+application directly serve iPXE artifacts so that systems can boot with no 
+other iPXE infrastructure required.
+
+## Motivation
+
+Booting a host using iPXE should be a first-class operation when using
+assisted-service. This has been found to be at least as common a method 
+as providing the live ISO via virtual media and is also required in some 
+environments. Users are already unpacking the ISO we provide into PXE 
+artifacts and serving those in their infrastructure. We can make this much 
+easier for them by serving those artifacts as well as the iPXE boot script 
+directly from our application.
+
+### Goals
+
+- Serve iPXE artifacts (kernel, rootfs, and initrd)
+- Serve an iPXE boot script
+- Include the ignition in the provided initrd
+
+### Non-Goals
+
+- Support PXE (tftp based)
+- Integration with ironic (separate effort)
+- iPXE boot script customization (maybe in a follow-up if it’s useful)
+- Configure the minimal ISO to pull the rootfs from the image service
+
+## Proposal
+
+The image-service will serve the kernel, rootfs, and initrd with ignition 
+appended (see Implementation Details), while the assisted service will 
+provide an API to download the boot script. The boot script created and served 
+by the assisted service will contain URLs to the required artifacts hosted by 
+the image service.
+
+### User Stories
+
+#### Story 1
+
+As a user of assisted installer I want to boot my hosts using iPXE without 
+setting up my own http server to host artifacts.
+
+### Implementation Details/Notes/Constraints
+
+#### Ironic Integration
+
+To support integration with Ironic’s existing iPXE boot methods, the 
+discovery ignition must be appended to the initrd rather than specified as 
+a separate kernel parameter. The image service will also handle this initrd 
+manipulation when an initrd is requested for a particular infra-env.
+
+#### New Image Service APIs
+
+Image service APIs will be added to download the required boot artifacts. The 
+rootfs and kernel will be served directly from the template image. The initrd 
+endpoint will be specific to an infra-env in order to embed the ignition.
+
+- `GET /boot-artifacts/{artifact-name}?{params}`
```

Comment:
is it under the same assisted swagger? how would it be exposed from c.rh.com?

Reply:
as far as i understand it's under the image service not assisted

Reply:
that's exactly my quesiton. How would it be served from c.rh.com if it's not under AI

Reply:
The image service is already exposed at a different path than assisted service.
ISO downloads from the production environment look like this today:
`https://api.openshift.com/api/assisted-images/images/{uuid}?{params}`

So this endpoint would be:
`https://api.openshift.com/api/assisted-images/boot-artifacts/{artifact-name}?{params}`

Reply:
But it doesn't matter - these URLs aren't really part of the API.  The URLs appear as properties in the infraenv.  Nobody should really be constructing them by hand.